### PR TITLE
chore: drop support for python 3.10 and django 3.2

### PR DESCRIPTION
**Pull request by magajh**
**Description**:

This PR removes support for Django 3.2 and Python 3.10 from our plugin.

**Rationale**:

We have decided to drop support for these versions because none of the Open edX releases we support use Django 3.2 or Python 3.10. Therefore, maintaining compatibility with them is unnecessary.

For more context and information, please refer to [AP-1490](https://edunext.atlassian.net/browse/AP-1490).

**Changes**:

- Updated the test workflow to remove Python 3.10 and Django 3.2 from the Python and Django matrix.
- Updated the `tox.ini` file to eliminate specifications for Python 3.10 and Django 3.2. 
- Deleted the `django32.txt` file.
- Removed constraints that were only necessary for compatibility with Python 3.10 or Django 3.2.
- Upgraded the requirements using Python 3.8, which is the oldest version we now support.

**How to Test**:

- Ensure that all tests in the GitHub CI pipeline pass successfully.
- Verify that the pipeline no longer runs tests with Python 3.10 and Django 3.2.

[AP-1490]: https://edunext.atlassian.net/browse/AP-1490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by magajh** [comment]
## Note for reviewers:
The integration test failures are **not related** with the changes in this PR. These errors will get fix in a different PR, similar to what was done for the rest of the plugins https://github.com/eduNEXT/eox-core/pull/288

**Review by BryanttV — DISMISSED** [review]
I agree with the changes in this PR. I would merge it until the workflow errors are corrected in another PR.

**Review by mariajgrimaldi — DISMISSED** [review]

**Review by MaferMazu — DISMISSED** [review]
Looks good. ✨

Limiting juniper and koa may be a good idea, but it is not a blocker because lilac is already limited.

**Comment by magajh** [comment]
> Looks good. ✨
> 
> Limiting juniper and koa may be a good idea, but it is not a blocker because lilac is already limited.

@MaferMazu should we limit Juniper and Koa with the same version as the limit in lilac v2.0.0?

**Comment by MaferMazu** [comment]
Yes, because after reviewing the versions, we have v1.0.0, and the following release is v2.0.0.
But you only need to add it in Koa because Juniper already has an upper limit.

**Review by mariajgrimaldi — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [v7.0.0](https://github.com/eduNEXT/eox-hooks/compare/v6.3.0...v7.0.0) - (2024-10-22)
```
Comment (BryanttV) [inline CHANGELOG.md:8]:
```suggestion
## [v7.0.0](https://github.com/eduNEXT/eox-hooks/compare/v6.3.0...v7.0.0) - (2024-11-20)
```

**Review by BryanttV — DISMISSED** [review]

**Review by mariajgrimaldi — APPROVED** [review]

**Review by BryanttV — APPROVED** [review]

**Pull request merged**